sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -641,6 +641,16 @@ object SQLConf {
     .intConf
     .createWithDefault(200)
 
+  val THRIFTSERVER_BATCH_DESERIALIZE_LIMIT =
+    buildConf("spark.sql.thriftServer.batchDeserializeLimit")
+      .doc("The maximum number of result rows that can be deserialized at one time. " +
+        "If the number of result rows exceeds this value, the Thrift Server will only use " +
+        "'memory of serialized rows' + 'memory of the deserialized rows being fetched to the " +
+        "client'. Only valid if spark.sql.thriftServer.incrementalCollect is false. " +
```

Comment:
nit: `s"client'. Only valid if ${THRIFTSERVER_INCREMENTAL_COLLECT.key} is false. " +`